python/ray/tests/test_failure.py

```diff
@@ -82,6 +82,38 @@ def f():
         assert False
 
 
+# Note that `ray_start_2_cpus` is required here. If we use
+# `ray_start_regular`, it's very likely that two tasks are submitted before
+# the first worker is registered to Raylet. Since
+# `maximum_startup_concurrency` is 1, worker pool will wait for the
+# registration of the first worker without starting any new workers. The
+# result is, the two tasks will be executed sequentially, which breaks an
+# assumption of this test case - the two tasks run in parallel.
+def test_get_throws_quickly_when_found_exception(ray_start_2_cpus):
+    @ray.remote
+    def bad_func1():
+        raise Exception("Test function intentionally failed.")
+
+    @ray.remote(max_retries=0)
+    def bad_func2():
+        eval("exit()")
+
+    @ray.remote
+    def slow_func():
+        time.sleep(10)
+
+    bad_functions = [bad_func1, bad_func2]
+    for bad_func in bad_functions:
+        objects = [bad_func.remote(), slow_func.remote()]
+        start_time = time.time()
+        with pytest.raises(ray.exceptions.RayError) as err:
+            ray.get(objects)
+        assert err.type is ray.exceptions.RayTaskError \
+            or err.type is ray.exceptions.RayWorkerError
+        duration = time.time() - start_time
+        assert duration < 5, "Should fail quickly."
```

Comment:
This looks like it's going to be flaky as well. Do we need the time check?

Reply:
It cannot verify the expected behavior if we don't check the time. How about make the `slow_func` sleep for a long time, like 1h, and add a timeout of 1m in `ray.get`. Then we assert no `RayTimeoutError` is thrown.

Reply:
Or we can just make these changes: `10 -> 3600` and `5 -> 60`. If we have the expected behavior, the `ray.get` will return in a short time. It's very unlikely that the `ray.get` returned without any exception but lasted for more than 1 minute.

Reply:
I've removed the time check.